### ISSUE TITLE
Notice of AGPL3 License Violation

### ISSUE THREAD

**Issue by quacksire** [Alex313031/apple-music-desktop#7]
It has come to our attention that your recent commit ([Commit Link](https://github.com/Alex313031/apple-music-desktop/commit/0ffb1c35c91d28d46612f1269790a17f4420e2d3#diff-c72a907ac323cd2f334ed0e2bd07d15ab62581c4753660c8a0d1c681b30be4b6)) blatantly violates points 4 and 5 of the GNU Affero General Public License v3.0, under which the Cider app is licensed.

We kindly remind you to ensure compliance with the terms of the AGPL3 license. Failure to do so may result in further actions to ensure the protection of our project's intellectual property rights.

**Comment by Alex313031** [comment]
@quacksire This doesn't use anything from Cider?

It uses stuff from apple-music-electron.

But in any case, how should I fix this?

**Comment by coredev-uk** [comment]
Hello,

Since Qwack is not replying to this, I will handle this on behalf of the team. 

If you could remove any reference to Cider in your codebase, notably the [`client-preload.js`](https://github.com/Alex313031/apple-music-desktop/blob/main/src/preload/client-preload.js) which has clearly been copied word by word from our version, including the logs referring to it, or if you continue to use this code please follow the AGPL license rules and adopt this license. 

As Cider has now been archived and I understand that there is not much on GitHub for Apple Music, this will not go any further than this. As far as I can see, the majority of the functionality you have copied, you have not even used. So I suggest removing it and rewriting as necessary.

Thank you for your cooperation.

**Comment by Alex313031** [comment]
@coredev-uk I will do both. Update the license and remove unneeded stuff from the preload script. I'm mainly just using a few musickit stuff from it to allow pausing/playing from the tray and menu.

**Comment by coredev-uk** [comment]
In that case, all you will need is to hook into PlaybackStateHasChanged (or WillChange) and/or NowPlayingHasChanged with just an IPC call to send the current playing item. 

If you want to know more about the types I suggest taking a look at our [types for MusicKit](https://github.com/ciderapp/musickit-types).

**Comment by Alex313031** [comment]
@coredev-uk Yeah, I looked at that, but I am still keeping the file around as client-preload.bak. It isn't used in the app anymore, or included in the installers. Just keeping it around as a backup reference, as looking through and seeing examples made it alot easier for me to come up with my own, simplified implementation, rather than reading through the apple docs (although the docs did help me with getting track info to make a new popup window to show the track name/album/artist when in docked/tray mode.

Should I still modify the license since the file is in the repo? Or can I exclude it since it isn't actually built into the product or installer.

**Comment by coredev-uk** [comment]
Just throw it in a docs folder outside the src and remove any direct references to cider, that should be all. As for apple-music-electron, its a very hard to keep it going since its highly dependant on Apple's site, hence the reason why we made Cider, just the paywalled nature of the application is a regrettable circumstance of our expanded user base.

I wish you good luck for this and hope it goes well and thanks again for following this.
